docs/design-proposals/node/Unified-Vertical-Scaling-for-Container-and-VM-Pods.md

```diff
@@ -0,0 +1,646 @@
+* If additional resize requests arrive when a Pod is being resized, those
+  requests are handled after completion of the resize that is in progress. And
+  resize is driven towards the latest desired state.
+* Lowering memory limits may not always take effect quickly if the application
+  is holding on to pages. Kubelet will use a control loop to set the memory
+  limits near usage in order to force a reclaim, and update the Pod's
+  Status.ContainerStatuses[i].Resources only when limit is at desired value.
+* Impact of Pod Overhead: Kubelet adds Pod Overhead to the resize request to
+  determine if in-place resize is possible.
+* Impact of memory-backed emptyDir volumes: If memory-backed emptyDir is in
+  use, Kubelet will clear out any files in emptyDir upon Container restart.
+* At this time, Vertical Pod Autoscaler should not be used with Horizontal Pod
+  Autoscaler on CPU, memory. This enhancement does not change that limitation.
+
+### Affected Components
+
+Pod v1 core API:
+* extended model,
+* modify RBAC bootstrap policy authorizing Node to update PodSpec,
+* extend NodeRestriction plugin limiting Node's update access to PodSpec only
+  to the ResourcesAllocated field,
+* new admission controller to limit update access to ResourcesAllocated field
+  only to Node, and mutates any updates to ResourcesAllocated & ResizePolicy
+  fields to maintain compatibility with older versions of clients,
+* added validation allowing only CPU and memory resource changes,
+* setting defaults for ResourcesAllocated and ResizePolicy fields.
+
+Admission Controllers: LimitRanger, ResourceQuota need to support Pod Updates:
+* for ResourceQuota, podEvaluator.Handler implementation is modified to allow
+  Pod updates, and verify that sum of Pod.Spec.Containers[i].Resources for all
+  Pods in the Namespace don't exceed quota,
+* PodResourceAllocation admission plugin is ordered before ResourceQuota.
+* for LimitRanger we check that a resize request does not violate the min and
+  max limits specified in LimitRange for the Pod's namespace.
+
+Kubelet:
+* set Pod's Status.ContainerStatuses[i].Resources for Containers upon placing
+  a new Pod on the Node,
+* update Pod's Spec.Containers[i].ResourcesAllocated upon resize,
+* change UpdateContainerResources CRI API to work for both Linux & Windows.
+
+Scheduler:
+* compute resource allocations using Pod.Spec.Containers[i].ResourcesAllocated.
+
+Other components:
+* check how the change of meaning of resource requests influence other
+  Kubernetes components.
+
+### Changes in Arktos-vm-runtime-service
+
+#### Background and context
+Currently in Arktos, the VM workload POD is defined the same way as container pod in terms of resources in the spec, i.e.
+
+    spec:
+      resource:
+       limits:
+         cpu:
+         memory:
+       request:
+         cpu:
+         memory:
+     
+Arktos controls the resource the same way as for containers, i.e. schedule based on requested resource level, and evict 
```

Comment:
You mean "controls **VM pod** resources the same way..."?

Reply:
yes, i will update the wording. thanks.